storage/store/inmemory/unsynchronized/collections.go

```diff
@@ -128,3 +130,26 @@ func (c *Collections) Remove(collID flow.Identifier) error {
 
 	return nil
 }
+
+// AddToBatch adds all the in-memory storages to the given batch.
+// It is used for the batching writes to the DB.
+func (c *Collections) AddToBatch(batch storage.ReaderBatchWriter) error {
+	writer := batch.Writer()
+
+	for _, coll := range c.collections {
+		light := coll.Light()
+		err := operation.UpsertCollection(writer, &light)
+		if err != nil {
+			return fmt.Errorf("could not persist collection: %w", err)
+		}
+	}
+
+	for _, coll := range c.lightCollections {
```

Comment:
this is storing the light collections a second time. I believe the only endpoint we call while indexing is `StoreLightAndIndexByTransaction`, so we need to store the light collections and `transactionIDToLightCollection` index.
https://github.com/onflow/flow-go/blob/93ce1c11bdde40e55c50a4c0696760069c42db9a/module/state_synchronization/indexer/indexer_core.go#L331-L370